tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+"tarball" (which has the `.tar.gz` file extension). The maintainers should also
+provide the files needed to verify the package -- in the case of Apache2, they
+do! Let's first download the tarball:
+
+```bash
+wget https://dlcdn.apache.org/httpd/httpd-2.4.62.tar.gz
+```
+
+#### Verifying the checksums
+
+For this particular version of `apache` there is no MD5 checksum file so we
+can't use the `md5sum` tool we used earlier. But, there *are* SHA256 and SHA512
+checksum files. Let's download the SHA256 checksum file -- it's advised to
+download these from the official source, so let's do that:
+
+```bash
+wget https://downloads.apache.org/httpd/httpd-2.4.62.tar.gz.sha256
+```
+
+Then we can verify the package download against that checksum using the
+`sha256sum` tool:
+
+```bash
+sha256sum -c httpd-2.4.62.tar.gz.sha256
+```
+
+Which should return:
+
+```bash
+httpd-2.4.62.tar.gz: OK
+```
+
+This tells us that the package file is validated. If the output returned
+`FAILED` then we would know something is wrong with it. In that case, we would
+want to download the file again (from a trusted source, preferably). We
+definitely wouldn't want to install it if the check fails! 
+
+In our case, since we obtained the package from a trusted source (the official
+website) and the checksums matched, we can proceed with the installation.
+
+#### Unpacking and installing
+
+First,  let's "un-tar" the tarball:
+
+```bash
+tar -xvzf httpd-2.4.62.tar.gz
+```
+
+This will unpack the contents of the tarball into a directory of the same name,
+which we can change to:
+
+```bash
+cd httpd-2.4.62
+```
+
+If we run `ls --all` we can see a list of all the files that are contained in
+the package.
+
+There are a couple of dependencies that we'll want to install that don't come
+packaged with the development version, so let's install those now:
+
+```bash
+sudo apt install autoconf libtool-bin
+```
+
+Next we'll want to configure the source tree -- the prefix is the most
+important option, so let's set that (for the sake of this example, we'll just
+set it to the default location):
+
+```bash
+./configure --prefix=/usr/local/apache2
+```
+
+Configuration is an important step, because this is where the Makefiles are
+created that will allow us to install the package. It will take a few moments
+as it checks all the files.
+
+Next, we want to compile by running `make`, and then install by running
+`make install`:
+
+```bash
+make
+make install
+```
+
+Now at this point, we could consider ourselves done, but installing this way
+means that there's no easy way to uninstall or update the package in the
+future since the package managers (`apt` and `dpkg`) can't track them. So one
+thing we can do is to create an installable deb package (which will help to
+make removal easier!) -- to do that, we can use the `fpm` utility.
+This is a Ruby package that needs to be installed via `gem`:
+
+```bash
+sudo apt install ruby ruby-dev gcc
+gem install fpm
+```
```

Comment:
I’m not a fan of using Ruby as the example for a couple reasons:
1. Ruby isn’t very popular anymore, so some users (particularly younger ones who weren’t connected to tech during the Ruby on Rails time) might not be familiar with it, and a more general example would be less distracting
2. Ruby’s package manager is called with `gem`, but Ruby also calls its packages “gems”, so it’s difficult for people brand new to this stuff to even Google what “ruby” or “gem” is

You could use an example with `pip` instead?

Reply:
In general, I'm ok with providing exposure to older tech (we even have content about Ruby in the server docs). In this case, we can't get `fpm` through `pip` so I'm a bit tied to Ruby here :/ (the alternatives you can get through pip are even more fiddly and annoying to use, so this is the least-worst option imo)